File: public/sdks/mxa_auth.py
import urllib.request
import urllib.error
import json
import platform
import subprocess

class MxaAuth:
    def __init__(self, secret: str, app_name: str, app_version: str):
        self.secret = secret
        self.app_name = app_name
        self.app_version = app_version
        self.base_url = "https://misanxauth.qzz.io"
        self.username = None
        self.subscription = None
        self.expiry = None

    def _rc4(self, key: str, data: str) -> str:
        s = list(range(256))
        j = 0
        res = []
        for i in range(256):
            j = (j + s[i] + ord(key[i % len(key)])) % 256
            s[i], s[j] = s[j], s[i]
        i = 0
        j = 0
        for char in data:
            i = (i + 1) % 256
            j = (j + s[i]) % 256
            s[i], s[j] = s[j], s[i]
            res.append(chr(ord(char) ^ s[(s[i] + s[j]) % 256]))
        return "".join(res)

    def _encrypt(self, data: str, key: str) -> str:
        cipher = self._rc4(key, data)
        return "".join(f"{ord(c):02x}" for c in cipher)

    def _decrypt(self, hex_data: str, key: str) -> str:
        data = "".join(chr(int(hex_data[i:i+2], 16)) for i in range(0, len(hex_data), 2))
        return self._rc4(key, data)

    def _get_hwid(self) -> str:
        try:
            if platform.system() == "Windows":
                cmd = "wmic csproduct get uuid"
                uuid = subprocess.check_output(cmd, shell=True).decode().split('\n')[1].strip()
                return uuid
            elif platform.system() == "Linux":
                with open("/etc/machine-id", "r") as f:
                    return f.read().strip()
            elif platform.system() == "Darwin":
                cmd = "ioreg -rd1 -c IOPlatformExpertDevice | grep -i UUID"
                uuid = subprocess.check_output(cmd, shell=True).decode().split('"')[-2]
                return uuid
        except Exception:
            pass
        return "UNKNOWN_HWID_" + platform.node()

    def _post(self, path: str, data: dict) -> dict:
        url = f"{self.base_url}{path}"
        headers = {"Content-Type": "application/json"}
        
        # Encrypt request payload
        plain_json = json.dumps(data)
        encrypted_payload = self._encrypt(plain_json, self.secret)
        
        wrapped_data = {
            "secret": self.secret,
            "payload": encrypted_payload
        }
        
        req_data = json.dumps(wrapped_data).encode("utf-8")
        req = urllib.request.Request(url, data=req_data, headers=headers, method="POST")
        try:
            with urllib.request.urlopen(req) as response:
                res_body = response.read().decode("utf-8")
                res_json = json.loads(res_body)
                
                # Decrypt response payload
                if "payload" in res_json:
                    decrypted_str = self._decrypt(res_json["payload"], self.secret)
                    return json.loads(decrypted_str)
                return res_json
        except urllib.error.HTTPError as e:
            try:
                res_body = e.read().decode("utf-8")
                res_json = json.loads(res_body)
                if "payload" in res_json:
                    decrypted_str = self._decrypt(res_json["payload"], self.secret)
                    return json.loads(decrypted_str)
                return res_json
            except Exception:
                return {"success": False, "message": f"HTTP_ERROR_{e.code}"}
        except Exception as e:
            return {"success": False, "message": f"CONNECTION_FAILED: {str(e)}"}

    def check_version(self) -> bool:
        resp = self._post("/versioncheck", {
            "secret": self.secret,
            "appName": self.app_name,
            "appVersion": self.app_version
        })
        return resp.get("success", False) and resp.get("message") == "VERSION_OK"

    def login(self, username: str, password: str) -> dict:
        resp = self._post("/login", {
            "username": username,
            "password": password,
            "secret": self.secret,
            "appName": self.app_name,
            "appVersion": self.app_version,
            "hwid": self._get_hwid()
        })
        if resp.get("success"):
            self.username = resp.get("username")
            self.subscription = resp.get("subscription")
            self.expiry = resp.get("expiry")
        return resp

    def register(self, username: str, password: str, license_key: str) -> dict:
        return self._post("/register", {
            "username": username,
            "password": password,
            "licenseKey": license_key,
            "secret": self.secret,
            "appName": self.app_name,
            "appVersion": self.app_version,
            "hwid": self._get_hwid()
        })

    def get_variable(self, var_name: str) -> str:
        resp = self._post("/getvariable", {
            "secret": self.secret,
            "appName": self.app_name,
            "appVersion": self.app_version,
            "varName": var_name
        })
        if resp.get("success"):
            return resp.get("value")
        return None

File: public/sdks/MxaAuth.cs
using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

public class MxaAuth
{
    public string Secret { get; }
    public string AppName { get; }
    public string AppVersion { get; }
    public string BaseUrl { get; set; } = "https://misanxauth.qzz.io";

    public string Username { get; private set; }
    public string Subscription { get; private set; }
    public string Expiry { get; private set; }

    private static readonly HttpClient client = new HttpClient();

    public MxaAuth(string secret, string appName, string appVersion)
    {
        Secret = secret;
        AppName = appName;
        AppVersion = appVersion;
    }

    private string Rc4(string key, string data)
    {
        int[] s = new int[256];
        for (int i = 0; i < 256; i++) s[i] = i;
        int j = 0;
        for (int i = 0; i < 256; i++)
        {
            j = (j + s[i] + key[i % key.Length]) % 256;
            int temp = s[i]; s[i] = s[j]; s[j] = temp;
        }
        int k = 0;
        int l = 0;
        StringBuilder res = new StringBuilder();
        for (int y = 0; y < data.Length; y++)
        {
            k = (k + 1) % 256;
            l = (l + s[k]) % 256;
            int temp = s[k]; s[k] = s[l]; s[l] = temp;
            res.Append((char)(data[y] ^ s[(s[k] + s[l]) % 256]));
        }
        return res.ToString();
    }

    private string Encrypt(string data, string key)
    {
        string cipher = Rc4(key, data);
        StringBuilder hex = new StringBuilder();
        for (int i = 0; i < cipher.Length; i++)
        {
            hex.Append(((int)cipher[i]).ToString("x2"));
        }
        return hex.ToString();
    }

    private string Decrypt(string hexData, string key)
    {
        StringBuilder data = new StringBuilder();
        for (int i = 0; i < hexData.Length; i += 2)
        {
            data.Append((char)Convert.ToInt32(hexData.Substring(i, 2), 16));
        }
        return Rc4(key, data.ToString());
    }

    private string GetHwid()
    {
        try
        {
            string hwid = Environment.MachineName + "-" + Environment.UserName + "-" + Environment.ProcessorCount;
            return hwid;
        }
        catch
        {
            return "UNKNOWN-HWID-" + Environment.MachineName;
        }
    }

    private async Task<ApiResponse> PostAsync(string path, object payload)
    {
        try
        {
            var plainJson = JsonSerializer.Serialize(payload);
            var encryptedPayload = Encrypt(plainJson, Secret);
            
            var wrapped = new
            {
                secret = Secret,
                payload = encryptedPayload
            };
            
            var json = JsonSerializer.Serialize(wrapped);
            var content = new StringContent(json, Encoding.UTF8, "application/json");
            var response = await client.PostAsync($"{BaseUrl}{path}", content);
            var responseJson = await response.Content.ReadAsStringAsync();
            
            using (JsonDocument doc = JsonDocument.Parse(responseJson))
            {
                var root = doc.RootElement;
                if (root.TryGetProperty("payload", out JsonElement payloadElement))
                {
                    string encryptedResponse = payloadElement.GetString();
                    string decryptedJson = Decrypt(encryptedResponse, Secret);
                    return JsonSerializer.Deserialize<ApiResponse>(decryptedJson, new JsonSerializerOptions
                    {
                        PropertyNameCaseInsensitive = true
                    });
                }
            }
            
            return JsonSerializer.Deserialize<ApiResponse>(responseJson, new JsonSerializerOptions
            {
                PropertyNameCaseInsensitive = true
            });
        }
        catch (Exception ex)
        {
            return new ApiResponse { Success = false, Message = "CONNECTION_ERROR: " + ex.Message };
        }
    }

    public async Task<bool> CheckVersionAsync()
    {
        var resp = await PostAsync("/versioncheck", new
        {
            secret = Secret,
            appName = AppName,
            appVersion = AppVersion
        });
        return resp != null && resp.Success && resp.Message == "VERSION_OK";
    }

    public async Task<ApiResponse> LoginAsync(string username, string password)
    {
        var resp = await PostAsync("/login", new
        {
            username = username,
            password = password,
            secret = Secret,
            appName = AppName,
            appVersion = AppVersion,
            hwid = GetHwid()
        });

        if (resp != null && resp.Success)
        {
            Username = resp.Username;
            Subscription = resp.Subscription;
            Expiry = resp.Expiry;
        }
        return resp;
    }

    public async Task<ApiResponse> RegisterAsync(string username, string password, string licenseKey)
    {
        return await PostAsync("/register", new
        {
            username = username,
            password = password,
            licenseKey = licenseKey,
            secret = Secret,
            appName = AppName,
            appVersion = AppVersion,
            hwid = GetHwid()
        });
    }

    public async Task<string> GetVariableAsync(string varName)
    {
        var resp = await PostAsync("/getvariable", new
        {
            secret = Secret,
            appName = AppName,
            appVersion = AppVersion,
            varName = varName
        });
        return resp != null && resp.Success ? resp.Value : null;
    }

    public class ApiResponse
    {
        public bool Success { get; set; }
        public string Message { get; set; }
        public string Username { get; set; }
        public string Subscription { get; set; }
        public string Expiry { get; set; }
        public string Value { get; set; }
    }
}
